### PR TITLE
ZCS-11426 Need new column in zimbra.volume table

### PR DESCRIPTION
**Pull request by inpati**
(cherry picked from commit ba8373da3c88fed0b04cbd73cf300878580aa8fb)

Added new colume 'store_type' in zimbra.volume table

### CODE REVIEW

File: src/db/migration/migrate20220525-Volume.pl [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+# Update Schema Version Number
+Migrate::updateSchemaVersion(111, 112);
+
+exit(0);
+
+# Function to add 'store_type' column
+sub addStoreTypeColumn() {
+    my $sql = <<VOLUME_ADD_COLUMN_EOF;
+ALTER TABLE volume ADD COLUMN IF NOT EXISTS store_type TINYINT NOT NULL DEFAULT '1' COMMENT '1 for onstore and 2 for s3 bucket';
```
Comment (deepakzimb) [inline src/db/migration/migrate20220525-Volume.pl:55]: As discussed with. you, we are using 1  as default value but in ticket it is 2, so correct the description in ticket.

File: src/db/migration/migrate20220525-Volume.pl [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/perl
+# 
+# ***** BEGIN LICENSE BLOCK *****
+# Zimbra Collaboration Suite Server
+# Copyright (C) 2005, 2007, 2008, 2009, 2010, 2013, 2014, 2016 Synacor, Inc.
```
Comment (deepakzimb) [inline src/db/migration/migrate20220525-Volume.pl:5]: correct copyright, should be 2022

**Review by deepakzimb — CHANGES_REQUESTED** [review]
I also want to know where we will include this pl file? zm-build repo is one place

@umagmrit please confirm on this

File: src/db/migration/migrate20220525-Volume.pl [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        sleep 1;
+    }
+
+    # prepare and execute query
+    my $sql = "SELECT path FROM volume";
+    my $sth = $dbh->prepare ($sql);
+    $sth->execute();
+
+    # print 'path' column from table volume
+    while(my @row = $sth->fetchrow_array()) {
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:79]: add space after while

File: src/db/migration/migrate20220525-Volume.pl [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+       printf("%s\t\n",$row[0]);
+       my $varBool = $PREFIX eq substr($row[0],0,length($PREFIX));
+       # printf(" --> do drive starts with prefix ? [%s]\n",$varBool);
+
+        if($varBool eq 1)
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:84]: add space after if

**Pull request closed without merging**